File: in C++/basic_ops.cpp
#include<iostream>
using namespace std;

class LinkedList{
    private:
        struct Node{
            int val;
            Node* next;

            Node(int value): val(value) , next(nullptr) {}
        };
        Node* head;

    public:
        LinkedList(): head(nullptr) {}

        void add(int val){
            Node* newNode = new Node(val);
            if (head == nullptr){
                head = newNode;
                return;
            }

            Node* temp = head;
            while (temp->next){
                temp = temp->next;
            }
            temp->next = newNode;
        }

        void display(){
            Node* temp = head;
            while (temp){
                cout<<temp->val<<" -> ";
                temp = temp->next;
            }
            cout<<"NULL"<<endl;
        }

        void deleteNode(int val) {
            if (head == nullptr) return;
        
            Node* current = head;
            Node* prev = nullptr;
        
            // Traverse until the node with the value is found
            while (current && current->val != val) {
                prev = current;
                current = current->next;
            }
        
            // Value not found
            if (current == nullptr) {
                cout << "Node with value " << val << " not found." << endl;
                return;
            }
        
            // Node to delete is the head
            if (current == head) {
                head = head->next;
                delete current;
                return;
            }
        
            // Node is in the middle or end
            prev->next = current->next;
            delete current;
        }        

        ~LinkedList(){
            Node* current = head;
            while (current){
                Node* nextNode = current->next;
                delete current;
                current = nextNode;
            }
        }
};

int main(){
    LinkedList l1;
    l1.add(1);
    l1.add(2);
    l1.add(3);
    l1.add(4);
    l1.add(5);
    l1.display();
    l1.deleteNode(6);
    l1.add(6);
    l1.display();
    return 0;
}

File: in Python/basic_ops.py
class Node:
    def __init__(self,data):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None
    
    def append(self, data):
        new_node = Node(data)

        if not self.head:
            self.head = new_node
            return
        
        last = self.head

        while last.next:
            last = last.next

        last.next = new_node

    def display(self):
        temp = self.head
        while temp:
            print(temp.data,end=" -> ")
            temp = temp.next
        print("None")

    def deleteNode(self, value):
        if self.head and self.head.data == value:
            self.head = self.head.next
            print(f"Node {value} delted")
            return
        temp = self.head
        while temp.next:
            if temp.next.data == value:
                temp.next = temp.next.next  
                print(f"Node {value} delted")
                return
            temp = temp.next
        print("Node not found")

l1 = LinkedList()
l1.append(1)
l1.append(2)
l1.append(3)
l1.append(4)
l1.append(5)
l1.display()
l1.deleteNode(1)
l1.deleteNode(3)
l1.deleteNode(5)
l1.display()
